token-metadata/example/src/processor.rs

```diff
@@ -0,0 +1,220 @@
+//! Program state processor
+
+use {
+    solana_program::{
+        account_info::{next_account_info, AccountInfo},
+        borsh::get_instance_packed_len,
+        entrypoint::ProgramResult,
+        msg,
+        program::set_return_data,
+        program_error::ProgramError,
+        program_option::COption,
+        pubkey::Pubkey,
+    },
+    spl_token_2022::{extension::StateWithExtensions, state::Mint},
+    spl_token_metadata_interface::{
+        error::TokenMetadataError,
+        instruction::{
+            Emit, Initialize, RemoveKey, TokenMetadataInstruction, UpdateAuthority, UpdateField,
+        },
+        state::{OptionalNonZeroPubkey, TokenMetadata},
+    },
+    spl_type_length_value::state::{
+        realloc_and_borsh_serialize, TlvState, TlvStateBorrowed, TlvStateMut,
+    },
+};
+
+fn check_update_authority(
+    update_authority_info: &AccountInfo,
+    expected_update_authority: &OptionalNonZeroPubkey,
+) -> Result<(), ProgramError> {
+    if !update_authority_info.is_signer {
+        return Err(ProgramError::MissingRequiredSignature);
+    }
+    let update_authority = Option::<Pubkey>::from(expected_update_authority.clone())
+        .ok_or(TokenMetadataError::ImmutableMetadata)?;
+    if update_authority != *update_authority_info.key {
+        return Err(TokenMetadataError::IncorrectUpdateAuthority.into());
+    }
+    Ok(())
+}
+
+/// Processes a [Initialize](enum.TokenMetadataInstruction.html) instruction.
+pub fn process_initialize(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    data: Initialize,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let metadata_info = next_account_info(account_info_iter)?;
+    let update_authority_info = next_account_info(account_info_iter)?;
+    let mint_info = next_account_info(account_info_iter)?;
+    let mint_authority_info = next_account_info(account_info_iter)?;
+
+    // scope the mint authority check, in case the mint is in the same account!
```

Comment:
Asking for clarification: Mint authority and mint in the same account you mean?

Reply:
yep! people do wacky things